### ISSUE TITLE
Loop (dev, 2.3.0 (57)) crashes on assert(isMutable) in CachedInsulinDeliveryObject+CoreDataClass

### ISSUE THREAD

**Issue by PieterGit** [LoopKit/Loop#1765]
**Describe the bug**
Loop crashes on assert(isMutable) in CachedInsulinDeliveryObject+CoreDataClass. Possible triggers:
* This might be caused by suspend/resuming the pump,
* Minmed500.PumpReservoirLow. Pump site was changed that evening
* Loop.loopNotLooping errors
* Lot's of pendingDoses (see issue report below)
* Some other reason in the pump history. 

Was first found on gitRevision a861aad (loop dev, Jun 25, 2022) but was also reproducable on latest dev 5e6ede3 (Aug 2, 2022)

**Attach an Issue Report**
Full Issue Report is available on request. Possible relevant part:

```
pumpRegion: Wereldwijd
reservoirUnits: Optional(97.875)
reservoirValidAt: Optional(2022-08-03 21:10:12 +0000)
unfinalizedBolus: nil
unfinalizedTempBasal: Optional(TempBasal rate:1.75 2022-08-03 21:11:36 +0000 duration:1800.0)
pendingDoses: [Suspend 2022-08-03 19:47:57 +0000, TempBasal rate:0.9 2022-08-03 19:44:40 +0000 duration:1502.9937930107117, TempBasal rate:1.6 2022-08-03 20:09:43 +0000 duration:298.98049199581146, TempBasal rate:0.325 2022-08-03 20:14:42 +0000 duration:597.96104991436, Bolus units:0.475 2022-08-03 20:12:30 +0000, TempBasal rate:0.0 2022-08-03 20:24:40 +0000 duration:602.0138560533524, TempBasal rate:1.45 2022-08-03 20:34:42 +0000 duration:24.030272006988525, Bolus units:1.0 2022-08-03 20:35:35 +0000, TempBasal rate:1.55 2022-08-03 20:35:06 +0000 duration:292.84264504909515, TempBasal rate:1.75 2022-08-03 20:39:59 +0000 duration:1468.6592049598694, TempBasal rate:1.75 2022-08-03 21:04:27 +0000 duration:46.53020703792572, TempBasal rate:1.65 2022-08-03 21:05:14 +0000 duration:21.69072198867798, TempBasal rate:1.6 2022-08-03 21:05:35 +0000 duration:38.15602695941925, TempBasal rate:1.65 2022-08-03 21:06:14 +0000 duration:12.156097054481506, Bolus units:0.8 2022-08-03 20:58:39 +0000, TempBasal rate:1.6 2022-08-03 21:06:26 +0000 duration:310.43721199035645]
timeZone: GMT+0200 (fixed)
```

**To Reproduce**
Uncertain. But after a bolus the app crashes. The bolus is given on app startup.

**Expected behavior**
That loop does not crash.

**Screenshots**
![image](https://user-images.githubusercontent.com/6500826/182728116-31b2e502-635a-4c3e-9262-a89fec0709c0.png)

**Phone**
 - Hardware: iPhone XR
 - OS Version: 15.4.1

**Loop Version**
 - Version Number: Loop Workspace Dev

**CGM**
 - Device: Dexcom G6
 - Manager app: Dexcom App

**Pump**
 - Manufacturer: Medtronic
 - Model: 754WW

**Additional context**
In the debugger I noticed a never ending object:
![image](https://user-images.githubusercontent.com/6500826/182728208-56a995f4-852c-4eef-b3ce-78cfb27af78e.png)

**Comment by PieterGit** [comment]
I saved several of the logfiles. If you need more information, let me know and I'll dig in to it. Thanks in advance.

**Comment by PieterGit** [comment]
In the console log I noticed two temp basals of 0.9 which startDate: 2022-08-03 19:44:40, but two different endDates: 19:46:22 and 19:46:22 . That seems weird.

```
2022-08-03 23:49:23.456438+0200 Loop[521:21848] [InsulinDeliveryStore] Add: DoseEntry(type: LoopKit.DoseType.bolus, startDate: 2022-08-03 19:40:04 +0000, endDate: 2022-08-03 19:40:44 +0000, value: 0.5, unit: LoopKit.DoseUnit.units, deliveredUnits: Optional(0.5), description: nil, insulinType: Optional(LoopKit.InsulinType.fiasp), automatic: Optional(false), manuallyEntered: false, syncIdentifier: Optional("9d1d27864291492fa820e704f0ef3117"), isMutable: false, wasProgrammedByPumpUI: false, scheduledBasalRate: nil)
2022-08-03 23:49:23.457117+0200 Loop[521:21848] [InsulinDeliveryStore] Update: DoseEntry(type: LoopKit.DoseType.tempBasal, startDate: 2022-08-03 19:44:40 +0000, endDate: 2022-08-03 19:46:22 +0000, value: 0.9, unit: LoopKit.DoseUnit.unitsPerHour, deliveredUnits: Optional(0.05), description: nil, insulinType: Optional(LoopKit.InsulinType.fiasp), automatic: nil, manuallyEntered: false, syncIdentifier: Optional("63011558db064998ac71c7475a46e8bc"), isMutable: false, wasProgrammedByPumpUI: false, scheduledBasalRate: Optional(0.68 IU/hr))
2022-08-03 23:49:23.458430+0200 Loop[521:21848] [InsulinDeliveryStore] Update: DoseEntry(type: LoopKit.DoseType.tempBasal, startDate: 2022-08-03 19:44:40 +0000, endDate: 2022-08-03 19:47:57 +0000, value: 0.9, unit: LoopKit.DoseUnit.unitsPerHour, deliveredUnits: Optional(0.05), description: nil, insulinType: Optional(LoopKit.InsulinType.fiasp), automatic: nil, manuallyEntered: false, syncIdentifier: Optional("63011558db064998ac71c7475a46e8bc"), isMutable: false, wasProgrammedByPumpUI: false, scheduledBasalRate: Optional(0.68 IU/hr))
LoopKit/CachedInsulinDeliveryObject+CoreDataClass.swift:238: Assertion failed
2022-08-03 23:49:23.459770+0200 Loop[521:21848] LoopKit/CachedInsulinDeliveryObject+CoreDataClass.swift:238: Assertion failed
2022-08-03 23:50:25.930466+0200 Loop[521:22310] [DeviceDataManager] PumpManager:MinimedPumpManager did fire heartbeat
(lldb)
```

**Comment by PieterGit** [comment]
Possibly relevent lines of the pump log:

```
Event: {
    "_type" = TempBasal;
    rate = "0.675";
    temp = absolute;
    timestamp = "2022-08-03T19:29:38Z";
    wasRemotelyTriggered = 1;
}
Event: {
    "_type" = TempBasalDuration;
    duration = 30;
    timestamp = "2022-08-03T19:29:38Z";
}
Event: {
    "_type" = BolusNormal;
    amount = "0.75";
    programmed = "0.75";
    timestamp = "2022-08-03T19:30:18Z";
    type = Normal;
    unabsorbed = "2.4";
    wasRemotelyTriggered = 1;
}
Event: {
    "_type" = TempBasal;
    rate = "0.375";
    temp = absolute;
    timestamp = "2022-08-03T19:34:37Z";
    wasRemotelyTriggered = 1;
}
Event: {
    "_type" = TempBasalDuration;
    duration = 30;
    timestamp = "2022-08-03T19:34:37Z";
}
Event: {
    "_type" = TempBasal;
    rate = "1.5";
    temp = absolute;
    timestamp = "2022-08-03T19:39:38Z";
    wasRemotelyTriggered = 1;
}
Event: {
    "_type" = TempBasalDuration;
    duration = 30;
    timestamp = "2022-08-03T19:39:38Z";
}
Event: {
    "_type" = BolusNormal;
    amount = "0.5";
    programmed = "0.5";
    timestamp = "2022-08-03T19:40:04Z";
    type = Normal;
    unabsorbed = "2.75";
    wasRemotelyTriggered = 1;
}
Event: {
    "_type" = TempBasal;
    rate = "0.9";
    temp = absolute;
    timestamp = "2022-08-03T19:44:38Z";
    wasRemotelyTriggered = 1;
}
Event: {
    "_type" = TempBasalDuration;
    duration = 30;
    timestamp = "2022-08-03T19:44:38Z";
}
Event: {
    "_type" = Suspend;
    timestamp = "2022-08-03T19:46:22Z";
    wasRemotelyTriggered = 0;
}
Event: {
    "_type" = Resume;
    timestamp = "2022-08-03T20:07:23Z";
    wasRemotelyTriggered = 0;
}
Event: {
    "_type" = Rewind;
    timestamp = "2022-08-03T20:07:38Z";
}
Event: {
    "_type" = Prime;
    amount = 2;
    primeType = manual;
    programmedAmount = 0;
    timestamp = "2022-08-03T20:08:20Z";
}
Event: {
    "_type" = Prime;
    amount = "0.3";
    primeType = fixed;
    programmedAmount = "0.3";
    timestamp = "2022-08-03T20:09:08Z";
}
````

**Comment by ps2** [comment]
This is fixed in Loop 3.0
